CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -0,0 +1,259 @@
+game:GetService("RunService"):BindToRenderStep("HopperBin3D", Enum.RenderPriority.Last.Value, function()
+	local cameraCFrame = workspace.CurrentCamera:GetRenderCFrame()
+	local cameraLook = cameraCFrame.lookVector
+	local cameraHorizontalVector = Vector3.new(cameraLook.X, 0, cameraLook.Z).unit
+
+	local cameraPitchAngle = math.asin(cameraLook.Y)
+	if cameraPitchAngle > math.rad(-25) then
+		local cameraHorizontalRotation = CFrame.new(zeroVector, cameraHorizontalVector)
+		horizontalRotation = cameraHorizontalRotation
+	end
+	
+	local position = workspace.CurrentCamera.CFrame.p	
+
+	local verticalError = math.abs((cameraPitchAngle - HOPPERBIN_ANGLE) / verticalRange)
+	local horizontalError = math.acos(cameraHorizontalVector:Dot(horizontalRotation.lookVector)) / horizontalRange
+	
+	SetTransparency(math.max(verticalError, horizontalError))
+	
+	local hopperbinVector = HOPPERBIN_ROTATION:vectorToWorldSpace(HOPPERBIN_OFFSET)
+	hopperbinVector = horizontalRotation:vectorToWorldSpace(hopperbinVector)
+	hopperbinPart.CFrame = CFrame.new(position + hopperbinVector, position)
+end)
+
+local image = Instance.new("ImageLabel", hopperbinGUI)
+image.Image = "http://www.roblox.com/asset/?id=286708119"
+image.Size = UDim2.new(0, 8, 0, 8)
+image.BackgroundTransparency = 1
+image.ZIndex = 2
+
+local mouse = game.Players.LocalPlayer:GetMouse()
+mouse.Icon = 'http://www.roblox.com/asset/?id=18662154'
+
+local uis = game:GetService("UserInputService")
+
+game:GetService("RunService"):BindToRenderStep("Cursor3D", Enum.RenderPriority.Last.Value, function()
+	if not game.Players.LocalPlayer.Character then
+		return
+	end
+
+	local cframe = workspace.CurrentCamera:GetRenderCFrame()
+	local ray = Ray.new(cframe.p, cframe.lookVector * 999)
+	local ignoreList = { game.Players.LocalPlayer.Character }
+	local part, endpoint = workspace:FindPartOnRayWithIgnoreList(ray, ignoreList)
+
+	image.Visible = false
+	if part ~= hopperbinPart then
+		return
+	end
+	image.Visible = true
+	
+	local localEndpoint = part:GetRenderCFrame():pointToObjectSpace(endpoint)
+	local x = ((localEndpoint.X / part.Size.X) * 1) + 0.5
+	local y = ((localEndpoint.Y / part.Size.Y) * 1) + 0.5
+	x = 1 - x
+	y = 1 - y
+	image.Position = UDim2.new(x, -image.AbsoluteSize.x * 0.5, y, -image.AbsoluteSize.y * 0.5)
+	
+	--REMOVE THIS WHEN GUI MOUSELEAVE/MOUSEENTER ARE FIXED
+	local px = image.AbsolutePosition.X + image.AbsoluteSize.X * 0.5
+	local py = image.AbsolutePosition.Y + image.AbsoluteSize.Y * 0.5
+	for i, v in pairs(Tools) do
+		v.OnLeave()
+		local ix = px - v.icon.AbsolutePosition.X
+		local iy = py - v.icon.AbsolutePosition.Y
+		if ix > 0 and ix < v.icon.AbsoluteSize.X and iy > 0 and iy < v.icon.AbsoluteSize.Y then
+			v.OnEnter()
+		end
+	end
+	------------------------------------------------------
+
+	uis.MouseBehavior = Enum.MouseBehavior.LockCenter
+	uis.MouseIconEnabled = false
+end)
+
+game:GetService("ContextActionService"):BindAction("HotbarEquip2", function(actionName, state, obj)
```

Comment:
Since this is becoming a core-script we need to use `BindCoreAction` instead of `BindAction`